File: app/src/main/java/com/roman/gurdan/sudo/pro/data/dao/GameDao.java
package com.roman.gurdan.sudo.pro.data.dao;

import androidx.annotation.NonNull;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.roman.gurdan.sudo.pro.data.entry.Game;

import java.util.List;

@Dao
public interface GameDao {

    @Query("select * from game")
    List<Game> getGames();

    @Update
    void update(@NonNull Game game);

    @Delete
    void delete(@NonNull Game game);

    @Insert
    void insert(@NonNull Game game);

    @Query("select count(id) from game")
    int getCount();

    @Query("select * from game where difficulty = :difficulty")
    List<Game> getGamesByDifficulty(int difficulty);

    @Query("select * from game where gameType = :type")
    List<Game> getGamesByType(int type);

    @Query("select count(id) from game where result = :result")
    int getGamesResultCount(int result);

    //近30天
    @Query("select * from game where DATE('YYYY-MM-DD', '-30 day') <= date group by date order by date DESC")
    List<Game> getGamesByDate();

    @Query("select count(id) from game where gameType = :gameType and date = :date")
    int getGames(int gameType, String date);

    @Query("select count(id) from game where gameType = :gameType and date = :date and result = :result")
    int getGames(int gameType, String date, int result);

}

File: app/src/main/java/com/roman/gurdan/sudo/pro/data/dao/GameDao.kt
package com.roman.gurdan.sudo.pro.data.dao

import androidx.annotation.NonNull
import androidx.room.*
import com.roman.gurdan.sudo.pro.data.entry.Game

@Dao
interface GameDao {

    @Query("select * from game")
    fun getGames(): List<Game>

    @Update
    fun update(@NonNull game: Game)

    @Delete
    fun delete(@NonNull game: Game)

    @Insert
    fun insert(@NonNull game: Game)

    @Query("select count(id) from game")
    fun getCount(): Int

    @Query("select * from game where difficulty = :difficulty")
    fun getGamesByDifficulty(difficulty: Int): List<Game>

    @Query("select * from game where gameType = :type")
    fun getGamesByType(type: Int): List<Game>

    @Query("select count(id) from game where result = :result")
    fun getGamesResultCount(result: Int): Int

    //近30天
    @Query("select * from game where DATE('yyyy-MM-dd', '-30 day') <= date group by date order by date DESC")
    fun getGamesByDate(): List<Game>

    @Query("select count(id) from game where gameType = :gameType and date = :date")
    fun getGames(gameType: Int, date: String): Int

    @Query("select count(id) from game where gameType = :gameType and date = :date and result = :result")
    fun getGames(gameType: Int, date: String, result: Int): Int

}